docs/book/src/plugins/plugins-versioning.md

```diff
@@ -0,0 +1,23 @@
+# Plugins Versioning
+Plugin version represents the version of an individual plugin, as well as the corresponding scaffolding that it generates. This version is defined in a plugin key, ex. `go.kubebuilder.io/v2`. See the [design doc](https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/plugins/extending-cli.md#plugin-versioning) for more details.
+
+## Incrementing plugin versions
+Changes made to plugins only require a plugin version increase if and only if a change is made to a plugin
+that breaks projects scaffolded with the previous plugin version. Once a plugin version `vX` is stabilized (it doesn't
+have an "alpha" or "beta" suffix), a new plugin package should be created containing a new plugin with version
+`v(X+1)-alpha`. Typically this is done by (semantically) `cp -r pkg/plugins/golang/vX pkg/plugins/golang/v(X+1)` then updating
+version numbers and paths. All further breaking changes to the plugin should be made in this package; the `vX`
+plugin would then be frozen to breaking changes.
+
+You must also add a migration guide to the [migrations](../migrations.md)
+section of the Kubebuilder book in your PR. It should detail the steps required
+for users to upgrade their projects from `vX` to `v(X+1)-alpha`.
+
+<aside class="note">
+<h1>Example</h1>
+
+Kubebuilder scaffolds projects with plugin `go.kubebuilder.io/v3` by default.
+
+You create a feature that adds a new marker to the file `main.go` scaffolded by `init` that `create api` will use to update that file. The changes introduced in your feature would cause errors if used with projects built with plugins `go.kubebuilder.io/v2` without users manually updating their projects. Thus, your changes introduce a breaking change to plugin `go.kubebuilder.io`, and can only be merged into plugin version `v3-alpha`. This plugin's package should exist already.
+
+</aside>
```

Comment:
We need +1 section : https://github.com/kubernetes-sigs/kubebuilder/blob/master/VERSIONING.md#versioning